website/static/js/filepage/index.js

```diff
@@ -17,6 +18,65 @@ var Panel = utils.Panel;
 
 var EDITORS = {'text': FileEditor};
 
+var MakeClient = function(element) {
+    makeClient(element);
+};
+
+var CopyButton = {
+    view: function(ctrl, params) {
+        return m('span.input-group-btn', m('button#copyBtn.btn.btn-default.btn-md[type="button"][style="height:' + params.copyButtonHeight + '"][data-clipboard-text="' + params.link + '"]',
+            {config: MakeClient},
+            m('.fa.fa-copy')));
+    }
+};
+
+var SharePopover =  {
+    view: function(ctrl, params) {
+        var copyButtonHeight = '34px';
+        var popoverWidth = '450px';
+        var link = params.link.substring(0, params.link.indexOf('download') + 8) + '%26mode=render';
+        var url = link.substring(0, link.indexOf('render'));
+        return m('button.btn.btn-sm.btn-primary.file-share', {onclick: function () {
+                m.render(document.getElementById('popOver'), [
+                    m('ul.nav.nav-tabs.nav-justified', [
+                        m('li.active', m('a[href="#share"][data-toggle="tab"]', 'Share')),
+                        m('li', m('a[href="#embed"][data-toggle="tab"]', 'Embed'))
+                    ]), m('br'),
+                    m('.tab-content', [
+                        m('.tab-pane.fade.in.active#share', m('.input-group', [
+                            m.component(CopyButton, {link: link, copyButtonHeight: copyButtonHeight}),
```

Comment:
Mithril doesn't seem to like that we're calling `m.render` in the view, because this `m.component` call isn't working correctly (the copy button doesn't display). One workaround would be to just call the `view` method directly:

```js
CopyButton.view(ctrl, {link: link, copyButtonHeight: copyButtonHeight}),
```

If you do this, please add a comment to explain why this is necessary.